lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java

```diff
@@ -31,9 +31,7 @@
   public interface LogicalTypeFactory {
     LogicalType fromSchema(Schema schema);
 
-    default String getTypeName() {
```

Comment:
I think you definitely need to keep the default for backwards compatibility. My suggestion just forces each implementing class to override the method - but it wouldn't break any current implementations of LogicalTypeFactory.

https://beginnersbook.com/2017/10/java-8-interface-changes-default-method-and-static-method/